### PR TITLE
Update rococo v1 to polakdot v0.9.1

### PR DESCRIPTION
**Pull request by samelamin**
some changes to get the collator node working and a parachain producing blocks

tested against polkadot v0.9.2

### CODE REVIEW

File: node/src/chain_spec.rs [hunk 1]
```diff
@@ -67,52 +97,65 @@ pub fn development_config(id: ParaId) -> ChainSpec {
...
+
+pub fn staging_test_net(id: ParaId) -> ChainSpec {
+	ChainSpec::from_genesis(
+		"Staging Testnet",
+		"staging_testnet",
+		ChainType::Live,
 		move || {
 			testnet_genesis(
-				get_account_id_from_seed::<sr25519::Public>("Alice"),
+				hex!["9ed7705e3c7da027ba0583a22a3212042f7e715d3c168ba14f1424e2bc111d00"].into(),
```
Comment (NukeManDan) [inline node/src/chain_spec.rs:130]: Why this? did you use a custom authority?

File: node/src/chain_spec.rs [hunk 2]
```diff
@@ -67,52 +97,65 @@ pub fn development_config(id: ParaId) -> ChainSpec {
...
 			)
 		},
-		vec![],
+		Vec::new(),
 		None,
 		None,
 		None,
 		Extensions {
-			relay_chain: "rococo-local".into(),
+			relay_chain: "westend-dev".into(),
```
Comment (NukeManDan) [inline node/src/chain_spec.rs:150]: This correct? you set rococo in the readme

File: node/src/chain_spec.rs [hunk 3]
```diff
@@ -132,5 +175,21 @@ fn testnet_genesis(
...
 		pallet_sudo: parachain_runtime::SudoConfig { key: root_key },
 		parachain_info: parachain_runtime::ParachainInfoConfig { parachain_id: id },
+		pallet_aura: parachain_runtime::AuraConfig {
+			authorities: initial_authorities,
+		},
+		cumulus_pallet_aura_ext: Default::default(),
+	}
+}
+
+fn shell_testnet_genesis(parachain_id: ParaId) -> cumulus_shell_runtime::GenesisConfig {
```
Comment (NukeManDan) [inline node/src/chain_spec.rs:185]: not needed for the template AFAIK

File: node/src/cli.rs [hunk 4]
```diff
@@ -1,6 +1,21 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
```
Comment (NukeManDan) [inline node/src/cli.rs:1]: I think we intentionally include no licenec - the "unlicense" should cover this

File: node/src/cli.rs [hunk 5]
```diff
@@ -51,15 +62,15 @@ pub struct ExportGenesisStateCommand {
 	/// Id of the parachain this state is for.
 	///
 	/// Default: 100
```
Comment (NukeManDan) [inline node/src/cli.rs:64]: 1000 now for non- system chains

File: node/src/cli.rs [hunk 6]
```diff
@@ -51,15 +62,15 @@ pub struct ExportGenesisStateCommand {
...
-	#[structopt(long, conflicts_with = "chain")]
```
Comment (NukeManDan) [inline node/src/cli.rs:54]: why remove thew struct op? and the next too

File: node/src/cli.rs [hunk 7]
```diff
@@ -81,9 +92,9 @@ pub struct ExportGenesisWasmCommand {
 
 #[derive(Debug, StructOpt)]
 #[structopt(settings = &[
-	structopt::clap::AppSettings::GlobalVersion,
-	structopt::clap::AppSettings::ArgsNegateSubcommands,
-	structopt::clap::AppSettings::SubcommandsNegateReqs,
+structopt::clap::AppSettings::GlobalVersion,
```
Comment (NukeManDan) [inline node/src/cli.rs:95]: nitpick - should include tab

File: node/src/command.rs [hunk 8]
```diff
@@ -1,21 +1,33 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
```
Comment (NukeManDan) [inline node/src/command.rs:1]: rm licence

File: node/src/command.rs [hunk 9]
```diff
@@ -24,18 +36,27 @@ fn load_spec(
 	id: &str,
 	para_id: ParaId,
 ) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
-	Ok(match id {
```
Comment (NukeManDan) [inline node/src/command.rs:27]: we want dev. Why change this? no staging or shell here in the template

File: node/src/command.rs [hunk 10]
```diff
@@ -24,18 +36,27 @@ fn load_spec(
...
+			} else {
+				Box::new(chain_spec)
+			}
+		}),
+	}
 }
 
 impl SubstrateCli for Cli {
 	fn impl_name() -> String {
-		"Parachain Collator Template".into()
```
Comment (NukeManDan) [inline node/src/command.rs:38]: keep this

File: node/src/command.rs [hunk 11]
```diff
@@ -44,7 +65,7 @@ impl SubstrateCli for Cli {
 
 	fn description() -> String {
 		format!(
-			"Parachain Collator Template\n\nThe command-line arguments provided first will be \
```
Comment (NukeManDan) [inline node/src/command.rs:47]: keep

File: node/src/command.rs [hunk 12]
```diff
@@ -57,45 +78,51 @@ impl SubstrateCli for Cli {
 	}
 
 	fn support_url() -> String {
-		"https://github.com/substrate-developer-hub/substrate-parachain-template/issues/new".into()
```
Comment (NukeManDan) [inline node/src/command.rs:60]: keep

File: node/src/command.rs [hunk 13]
```diff
@@ -57,45 +78,51 @@ impl SubstrateCli for Cli {
...
+		"https://github.com/paritytech/cumulus/issues/new".into()
 	}
 
 	fn copyright_start_year() -> i32 {
 		2017
 	}
 
 	fn load_spec(&self, id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
-		load_spec(id, self.run.parachain_id.unwrap_or(200).into())
+		load_spec(id, self.run.parachain_id.unwrap_or(100).into())
```
Comment (NukeManDan) [inline node/src/command.rs:89]:
```suggestion
		load_spec(id, self.run.parachain_id.unwrap_or(1000).into())
```

File: node/src/command.rs [hunk 14]
```diff
@@ -57,45 +78,51 @@ impl SubstrateCli for Cli {
...
 	}
 
-	fn native_runtime_version(_: &Box<dyn ChainSpec>) -> &'static RuntimeVersion {
```
Comment (NukeManDan) [inline node/src/command.rs:71]: keep

Comment (NukeManDan) [reply node/src/command.rs:71]: the rest of the file needs to remain for the template

File: runtime/Cargo.toml [hunk 15]
```diff
@@ -46,6 +47,7 @@ frame-system-benchmarking = { git = 'https://github.com/paritytech/substrate.git
 frame-system-rpc-runtime-api = { git = 'https://github.com/paritytech/substrate.git', branch = 'polkadot-v0.9.2', default-features = false }
 
 ## Substrate Pallet Dependencies
+pallet-assets = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.2" }
```
Comment (NukeManDan) [inline runtime/Cargo.toml:50]: not used?

File: runtime/Cargo.toml [hunk 16]
```diff
@@ -58,10 +60,12 @@ pallet-transaction-payment-rpc-runtime-api = { git = 'https://github.com/parityt
...
 cumulus-pallet-aura-ext = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
 cumulus-pallet-parachain-system = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
-cumulus-primitives-core = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
+cumulus-pallet-dmp-queue  = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
+
 cumulus-pallet-xcm = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
-cumulus-pallet-dmp-queue = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
 cumulus-pallet-xcmp-queue = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
+cumulus-ping = { git = 'https://github.com/paritytech/cumulus.git', branch = 'polkadot-v0.9.2', default-features = false }
```
Comment (NukeManDan) [inline runtime/Cargo.toml:67]: not used?

File: runtime/src/lib.rs [hunk 17]
```diff
@@ -204,57 +203,83 @@ parameter_types! {
 		})
 		.avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
 		.build_or_panic();
-	pub const SS58Prefix: u8 = 42;
+	pub const SS58Prefix: u8 = 0;
```
Comment (NukeManDan) [inline runtime/src/lib.rs:206]:
```suggestion
	pub const SS58Prefix: u8 = 42;
```
use the right one ;)

**Review by NukeManDan — CHANGES_REQUESTED** [review]
Thanks for this!! There are some things that need to be manually translated to keep this as a template, not a copy of cumulus. If you like, make some changes here to fix, but I intend to implement these myself anyhow to learn 😁 
LMK.

**Comment by samelamin** [comment]
@NukeManDan no worries, I tried to keep it as close to cumulus to get it to build. I will work on these changes tomorrow, now that its built all these changes should be relatively simple to implement 👍

**Comment by NukeManDan** [comment]
Thanks again for the PR! I think I got my PR working now, with the updates needed. 🙏🏼 
I am going to close this in favor of that for now.

**Pull request closed without merging**